spec/socialprovider.spec.js

```diff
@@ -102,6 +102,19 @@ describe("Tests for message batching in Social provider", function() {
     expect(xmppSocialProvider.client.send).toHaveBeenCalled();
   });
 
+  it("sends message to correct destinations", function() {
+    xmppSocialProvider.sendMessage('Bob', 'Hi', function() {});
+    xmppSocialProvider.sendMessage('Alice', 'Hi', function() {});
+    expect(xmppSocialProvider.client.send).not.toHaveBeenCalled();
+    jasmine.clock().tick(100);
+    expect(xmppSocialProvider.client.send.calls.count()).toEqual(2);
+    var dest = xmppSocialProvider.client.send.calls.first().args[0].parent.attrs.to;
+    if (dest !== 'Bob' && dest !== 'Alice') {
+      expect(dest).toEqual('Bob');
```

Comment:
OK, this code is fine then, but can you add a comment explaining that your  ```expect(dest).toEqual('Bob');``` is intended to fail as dest should have been 'Bob' or 'Alice'?